src/AzureAppConfigurationImpl.ts

```diff
@@ -490,11 +478,20 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
             return Promise.resolve(false);
         }
 
-        // check if any feature flag is changed
+        const needRefresh = await this.#checkKeyValueCollectionChanged(this.#featureFlagSelectors);
+        if (needRefresh) {
+            await this.#loadFeatureFlags();
+        }
+
+        this.#featureFlagRefreshTimer.reset();
+        return Promise.resolve(needRefresh);
+    }
+
+    async #checkKeyValueCollectionChanged(selectors: PagedSettingSelector[]): Promise<boolean> {

```

Comment:
In JS provider code implementation, there're many terms like `loadFeatureFlags`, `loadSelectedAndWatchedKeyValues`, `keyValueSelectors`, `featureFlagSelectors` ... are used for key values and feauture flags these two kinds settings.  Here `KeyValueCollection`  term make people confused what  kind of settings it checked. And it's strange when `featureFlagSelectors` are passed as parameter to this function. Besides, `ConfigurationSettings` is used in line239 when loaded setting can be both key values or feature flags. Let's keep good programming practice. Rename this function (like `checkConfiguraionSettingsChanged`) will bring our code base better Reusability, Readability, Maintainability.

Reply:
`checkConfiguraionSettingsChanged` sounds good. Updated. 

The reason I used the name `key value collection` is because I refered .net provider
![image](https://github.com/user-attachments/assets/55b540c1-1ef1-454b-af1e-15f19d1eed21)